src/index/MmapBasedArray.h

```diff
@@ -0,0 +1,190 @@
+// Copyright 2018, University of Freiburg,
```

Comment:
I think this belongs in `util` because it is quite generic. Also it should be namespaced and "Based" is redundant so just name it `MmapArray` or better yet since it can grow `MmapVector`